cfe_internal/enterprise/mission_portal.cf

```diff
@@ -129,18 +130,39 @@ bundle agent mission_portal_apache_from_stage(config, staged_config)
 
   commands:
 
-    mission_portal_apache_config_repaired.!systemd_supervised::
+    !systemd_supervised::
       "LD_LIBRARY_PATH=$(sys.workdir)/lib:$LD_LIBRARY_PATH $(sys.workdir)/httpd/bin/apachectl"
-        args => "restart",
-        classes => kept_successful_command,
+        args => "stop",
+        if => and( returnszero("$(validate_config) > /dev/null 2>&1 ", "useshell"),
+                   isnewerthan( $(staged_config), $(config) ) ),
         contain => in_shell,
-        comment => "We have to restart apache after a config change in order
-                    for the changes to take effect.";
+        classes => results( "apache_stop_after_new_staged_config" ),
```

Comment:
I tried an MPF upgrade from 3.15.4 to this and got this error. Do you need the first arg to be `"bundle"` here in the `results` function calls?

```
   error: Argument arity mismatch in body 'results' at line 154 in file '/var/cfengine/inputs/cfe_internal/enterprise/mission_portal.cf', expected 2, got 1
```

Reply:
good catch, there is another one of these a bit further down.